### ISSUE TITLE
Unable To run

### ISSUE THREAD

**Issue by ffzzc7** [Manca/face-detection#5]
Hi I'm getting following error
??? Attempt to reference field of non-structure array.

Error in ==> main at 80
all_features = [positiveFeatures.positiveFeatures; negativeFeatures.negativeFeatures];

**Comment by Manca** [comment]
Hi there,

Did you get the training set and run the training algorithm before running the main program?

**Comment by ffzzc7** [comment]
Yeah, I downloaded the  MIT set of training samples and put it in "mit\train\faces" and "mit\train\non-face"directory. But, I don't know how to run the training algorithm. Would you mind telling me how to do it?

**Comment by Manca** [comment]
Well, by running main.m for the first time the program should figure out that there's no training set and it'll generate positive and negative features from the training images and then run the training algorithm automatically.
For the training you'll need GML_AdaBoost_Matlab_Toolbox_0.3 and you'll need to edit line 11 in main.m to include path to your copy of GML AdaBoost toolbox.

That should be it. Let me know if you still have issues and post the errors that you get.

Hope it helps.

**Comment by ffzzc7** [comment]
Thank you very much for your help!
 I have adjusted the line 11 in main with my GML AdaBoost installation path before I ran the main.m and then I got the error which I mentioned in my first messege.

I have trouble with understanding the code below and this code is recognized as an error by the matlab praograme.
`all_features = [positiveFeatures.positiveFeatures; negativeFeatures.negativeFeatures];`(main at 81).
The variable "positiveFeatures.positiveFeatures" seems like a structrue variable.
 It will be so nice of  you if you can explain the meaning of this code.

**Comment by Manca** [comment]
When you run the program do you get the following messages:

"Extracting features from positive image"
"Extracting features from negative image"

If not, you might want to check if the training set is correctly setup.

Expected path of images is:

mit\train\face\face0... - for positive faces
and
mit\train\non-face\B1_0... - for negative faces

Let me know if that's setup correctly.

**Comment by ffzzc7** [comment]
The command window display this messages:
"Extracting features from positive image: 100
Extracting features from negative image: 100
Extracting features from negative image: 200"
and the next line is error hint.
So I guess the training set should be correctly setup.

**Comment by Manca** [comment]
After you run the program, do you have: posFeatures.mat and negFeatures.mat in your working folder?

If you do, what happens when you try to run:

positiveFeatures = load('posFeatures.mat');

from within Matlab command line?

**Comment by ffzzc7** [comment]
I don't have posFeatures.mat and negFeature,mat, instead I have negFeaturesBig.mat and posFeaturesBig.mat.

**Comment by Manca** [comment]
Ok, thanks.

Can you try this:

main.m, line 47
change to: 
    positiveFeatures = load('posFeaturesBig.mat');

and main.m line 74
change to:
    negativeFeatures = load('negFeaturesBig.mat');

OR simply rename your negFeaturesBig.mat to negFeatures.mat and posFeaturesBig.mat to posFeatures.mat without code changes.

This should work... Let me know.

**Comment by ffzzc7** [comment]
Sorry to bother you again. Still not working...
The command window shows:

Training starts...Current time: 14:4:42
??? Error using ==> times
Matrix dimensions must agree.

Error in ==> tree_node_w.train at 31
left_pos  = sum((calc_output(left , dataset) == labels).* weights);

Error in ==> RealAdaBoost at 60
  nodes = train(WeakLrn, Data, Labels, distr);

Error in ==> main at 92
  [learners weights] = RealAdaBoost(weak_learner, all_features, labels, 100);
 
But, I think the argument should be ok. "all_features" is <35802x400 double> matrix and "lables" is <1x400 double> matrix.
Any idea?

**Comment by mittalrajat** [comment]
I am getting the same error as well.

Any suggestions?

**Comment by KunWang123** [comment]
main line 91; RealAdaBoost line 60; train line 31 error use .*;
have you met this question?

**Comment by xiaobudianch** [comment]
I ran this program, and I had the same problem：
main
Extracting features from positive image: 100
Extracting features from negative image: 100
Extracting features from negative image: 200
Attempt to reference field of non-structure array.

Error in main (line 80)
all_features = [positiveFeatures.positiveFeatures; negativeFeatures.negativeFeatures];

According to your method, it cannot be solved. Is there any other way
